packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -0,0 +1,213 @@
+/*!
+ * Copyright 2020 Palantir Technologies, Inc.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context, RuleModule } from "@monorepolint/core";
+import { mutateJson, PackageJson } from "@monorepolint/utils";
+import path from "path";
+import * as r from "runtypes";
+import { coerce } from "semver";
+import { IPackageDependencyGraphNode, PackageDependencyGraphService } from "./util/packageDependencyGraphService";
+
+const Options = r.Union(
+  r.Partial({
+    skipUnparseableRanges: r.Undefined,
```

Comment:
should we have an option to also only consider in-repo packages?

Reply:
can we also have an option to specify exactly which peer deps? i'd imagine we violate a _LOT_ of peer dependencies, and we'd need to tackle them one at a time (but I mostly care about react right now, but i like that this rule is generic)

Reply:
> should we have an option to also only consider in-repo packages?

this currently checks that each in-repo package has correct peer dependencies (based on direct dependencies, some of which may be external libs).

are you suggesting we add an option to only check compatibility against internal libs? I'd rather wait and add that only if we realize that a lot of external libs have ridiculous peer declarations.

Reply:
> can we also have an option to specify exactly which peer deps? i'd imagine we violate a _LOT_ of peer dependencies, and we'd need to tackle them one at a time (but I mostly care about react right now, but i like that this rule is generic)

must've been typing at the same time 😄
see https://github.com/monorepolint/monorepolint/pull/116#discussion_r409090515

Reply:
you can disregard my first comment, i think i misread the code when i typed that.